youtube_dl/extractor/giphy.py

```diff
@@ -0,0 +1,61 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    remove_end,
+    remove_start,
+)
+
+
+class GiphyIE(InfoExtractor):
+    _VALID_URL = r'https://(?:i|media)\.giphy\.com/media/(\w+)/|https://giphy\.com/gifs/[^/#?]+-(\w+)(?:[#?]|$)'
+    _TESTS = [{
+        'url': 'https://media.giphy.com/media/2rj8VysAig8QE/giphy.gif',
+        'md5': 'faa04a4e3134cb74005b78b28c34cc29',
+        'info_dict': {
+            'id': '2rj8VysAig8QE',
+            'ext': 'mp4',
+            'title': 'Css GIF',
+        }
+    }, {
+        'url': 'https://i.giphy.com/media/oY7zuBsNna7CM/giphy.mp4',
+        'only_matching': True
+    }, {
+        'url': 'https://giphy.com/gifs/quit-playin-MZAvNpYCrFUJi',
+        'only_matching': True
+    }]
+
+    def _real_extract(self, url):
+        [video_id] = filter(None, re.match(self._VALID_URL, url).groups())
```

Comment:
What's the point of all this mess? This can be easily expressed with a single capture group in regex.

Reply:
You are not being helpful.